lib/ledger.rb

```diff
@@ -0,0 +1,47 @@
+class Ledger
+  attr_accessor :transactions, :balance
+
+  def initialize
+    @transactions = []
+    @balance = 0
+  end
+
+  def print_statement
+    print_heading
+    create_transaction_history
+  end
+
+  private
+
+  def print_heading
+    puts 'date || credit || debit || balance'
+  end
+
+  def create_transaction_history
+    history = []
+    sort_transactions.each do |transaction|
+      history << if transaction[:type] == :debit
```

Comment:
Is it possible to remove this conditional with polymorphism? (This would require writing classes that represent a view of a line of a statement.